client/src/pages/SignUp.js

```diff
@@ -0,0 +1,346 @@
+    });
+    await this.setState({
+      lastNameMessage:
+        this.state.lastName === "" ? "Last name is required!" : "",
+    });
+    await this.setState({
+      emailMessage: /^\S+@\S+\.\S+$/.test(this.state.email)
+        ? ""
+        : "Invalid email!",
+    });
+    await this.setState({
+      passwordMessage:
+        this.state.password.length > 5
+          ? ""
+          : "Password needs 6 characters or more!",
+    });
+    await this.setState({
+      passwordConfirmedMessage:
+        this.state.password === this.state.passwordConfirmed
+          ? ""
+          : "Both passwords are needed to be the same!",
+    });
+    var success = false;
+    if (
+      this.state.firstNameMessage === "" &&
+      this.state.lastNameMessage === "" &&
+      this.state.emailMessage === "" &&
+      this.state.passwordMessage === "" &&
+      this.state.passwordConfirmedMessage === ""
+    ) {
+      success = true;
+    }
+    await this.setState({
+      severity: success ? "success" : "error",
+      successMessage: success ? "Sign up successfully!" : "",
+    });
+    if (success === true) {
+      //fetch here
+
+      const res = fetch("http://localhost:3001/signup", {
+        method: "POST",
+        headers: { "Content-Type": "application/json"},
+        body: JSON.stringify({
+          firstName: this.state.firstName,
+          lastName: this.state.lastName,
+          email: this.state.email,
+          password: this.state.password,
+          confirmPassword: this.state.passwordConfirmed
+        }),
+      })
+        .then((response) => response.json())
+        .then((responseJson) => {
+          return responseJson.errors;
+        })
+        .catch((error) => {
+          console.error(error);
+        });
+
+/*
+      axios
+        .post("http://localhost:3001/signup", {
+          first_name: this.state.firstName,
+          last_name: this.state.lastName,
+          email: this.state.email,
+          password: this.state.password,
+          confirmPassword: this.state.passwordConfirmed
+        })
+        .then((response) => {
+          this.setState(
+            {
+              //      successMessage: response.data.msg
+              successMessage: response.status,
+            },
+            () => {}
+          )
+          .catch(function (error) {
+            console.log(error);
+          });
+        });
+        */
+    }
+  }
+
+  handleClose = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    this.setState({ openSnack: false });
+  };
+
+  render() {
+    const { classes } = this.props;
+    return (
+      <div className={classes.root}>
+        <Paper className={classes.paper} elevation={0}>
+          <Grid class="frame" container spacing={0}>
+            <Grid item>
+              <Grid class="image">
+                <img className={classes.img} alt="complex" src={(width < 680) ? pic2 : (width > 1100) ? pic1 : pic3}/>
```

Comment:
There is a lot of repeated code in Signup and Login - potentially you could make one reusable component that is used in both components and make the forms another component?

Reply:
> There is a lot of repeated code in Signup and Login - potentially you could make one reusable component that is used in both components and make the forms another component?

I am planning to take the image and its CSS into a new file and the right top button into another new file. Is it good?

Reply:
I combined both files into one to reuse the common codes.